compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -1184,7 +1184,9 @@ public void visit(BLangTableConstructorExpr tableConstructorExpr, AnalyzerData d
             }
             data.resultType = symTable.semanticError;
         } else {
-            data.resultType = symTable.semanticError;
+            data.resultType = types.checkType(tableConstructorExpr.pos,
```

Comment:
Changed

Reply:
@SandaruJayawardana, I checked this and this change does not work because we are getting unwanted errors. Please check following example,

```ballerina
CusTable tab1 = table key(id, firstName) [
    { id: 222, firstName: "Sanjiva", lastName: "Weerawarana" },
    { id: 111, firstName: "James", lastName: "Clark" }
];
```
In this example the only error we should get is `unknown type 'CusTable'`. But with your suggestion, we are getting one more incompatible types error which is not needed. Also, there are several cases we are getting unnecessary changes. The `checkType` method handles this properly, So I think we need this.